nginx.conf

```diff
@@ -52,13 +52,20 @@ http {
       proxy_pass http://localhost:8383;
       proxy_redirect off;
 
-      add_header Set-Cookie $session_cookie;
-      proxy_set_header X-Forwarded-Proto https;
-
       # buffer requests, but not responses, so streaming out works.
       proxy_request_buffering on;
       proxy_buffering off;
       proxy_read_timeout 2m;
+
+      # Dev-specific hacks:
+
+      # Remap "Set-Cookie: __Host-session=..." to "Set-Cookie: session=..."
+      # TODO but why is this helpful?
+      add_header Set-Cookie $session_cookie;
+
+      # Trick central-backend from thinking connections are coming
+      # over HTTPS so that ExpressJS will set "secure" cookies.
```

Comment:
I'm not sure whether we need Express to set Secure cookies in local development, but if we didn't specify `X-Forwarded-Proto`, any attempt to use cookie auth would fail this Backend check: https://github.com/getodk/central-backend/blob/5aa8ff4a4e91a65574b19774e655a54c36999209/lib/http/preprocessors.js#L99

Reply:
Aha!  I suspect that check may technically be redundant due to a combination of:

1. https://github.com/getodk/central-backend/blob/master/lib/resources/sessions.js#L42:

```js
response.cookie('__Host-session', session.token, { path: '/', expires: session.expiresAt,
  httpOnly: true, secure: true, sameSite: 'strict' });
```

2. https://expressjs.com/en/4x/api.html#res.cookie: 



Property | Type | Description
-- | -- | --
`secure` | Boolean | Marks the cookie to be used with HTTPS only.

3. https://expressjs.com/en/4x/api.html#req.protocol

> ### `req.protocol`
>
> Contains the request protocol string: either http or (for TLS requests) https.
>
> When the [trust proxy setting](https://expressjs.com/en/4x/api.html#trust.proxy.options.table) does not evaluate to false, this property will use the value of the X-Forwarded-Proto header field if present. This header can be set by the client or by the proxy.

Reply:
Interesting. It seems like (3) means that the two conditions in the check could be identical. And (1) and (2) mean that the check as a whole shouldn't be needed. My instinct is to leave Backend as-is for now and just document how all this is working, but we could also consider removing the check from Backend.

Reply:
What do you think about "Trick central-backend from thinking connections are coming over HTTPS so that it will use the session cookie." ? I believe that Chrome and Firefox are actually both happy to accept a Secure cookie over http://localhost (maybe Safari too these days, not sure).

Reply:
With less assumed knowledge about the project, I prefer the current reading.